controllers/workspace/provision/deployment.go

```diff
@@ -35,10 +36,16 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
 	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+const (
+	ContainerCrashLoopBackOffReason = "CrashLoopBackOff"
```

Comment:
It works for me when I do `sleep 1` but when I do `exit 1`, it does not work:
```
kind: DevWorkspace
apiVersion: workspace.devfile.io/v1alpha1
metadata:
  name: web-terminal
  annotations:
    controller.devfile.io/restricted-access: "true"
  labels:
    # it's a label OpenShift console uses a flag to mark terminal's workspaces
    console.openshift.io/terminal: "true"
spec:
  started: true
  routingClass: 'web-terminal'
  template:
    components:
      - plugin:
          name: web-terminal
          id: redhat-developer/web-terminal/latest
      - container:
          memoryLimit: "256Mi"
          name: tooling
          image: quay.io/wto/web-terminal-tooling:latest
          command: ["exit"]
          args: ["1"]
          mountSources: false
          args: ["tail", "-f", "/dev/null"]
          env:
            - name: PS1
              value: \[\e[34m\]>\[\e[m\]\[\e[33m\]>\[\e[m\]
          volumeMounts:
            - name: home-storage
              path: "/home/user"
      - volume:
          name: home-storage
```

```
  containerStatuses:
  - image: quay.io/wto/web-terminal-tooling:latest
    imageID: ""
    lastState: {}
    name: tooling
    ready: false
    restartCount: 0
    started: false
    state:
      waiting:
        message: |
          container create failed: time="2021-04-20T11:54:11Z" level=error msg="container_linux.go:366: starting container process caused: exec: \"exit\": executable file not found in $PATH"
        reason: CreateContainerError

```

I'm fully OK to move it out of the current PR scope to move forward.

Reply:
> I tested and it works fine for me. Good job sorry for bothering with tiny refactoring stuff.

@sleshchenko Thanks for the review! It's not a bother :-) It looks much better. I've incorporated all of the suggestions above.

Ohhh okay, i was using exit 1 this way and it seems to work: 
```
kind: DevWorkspace
apiVersion: workspace.devfile.io/v1alpha2
metadata:
  name: plain
spec:
  started: true
  routingClass: 'basic'
  template:
    components:
      - name: nodejs
        container:
          image: quay.io/wto/web-terminal-tooling:latest
          memoryLimit: 512Mi
          mountSources: true
          command:
            - "/bin/bash"
          args:
            - "exit 1"
```

I'll work on figuring out why the sample you provided above doesn't work. I don't mind doing it as part of this PR, it makes sense that it should work for the sample you provided.

Reply:
Seems I see the difference: I missed removing one args, so it probably got `exit tail -f /dev/null`.

Reply:
BTW it's just one more of the possible errors we can propagate to the user.

Reply:
fixed! 
![image](https://user-images.githubusercontent.com/40298444/115419036-fff8ee80-a1c7-11eb-9e36-62f88c64c553.png)